xpti/CMakeLists.txt

```diff
@@ -79,10 +79,7 @@ add_subdirectory(src)
 if (LLVM_BINARY_DIR)
   file(GLOB_RECURSE XPTI_HEADERS_LIST CONFIGURE_DEPENDS "${CMAKE_CURRENT_SOURCE_DIR}/include/xpti/*")
   add_custom_target(xpti-headers
-    DEPENDS ${LLVM_BINARY_DIR}/include/xpti)
-
-  add_custom_command(
-    OUTPUT  ${LLVM_BINARY_DIR}/include/xpti
+    BYPRODUCTS ${LLVM_BINARY_DIR}/include/xpti
```

Comment:
Could you please double-check (using Cmake 3.14) the cases when:
 - project is rebuilt without changes in xpti headers - no dependent targets are built
 - the project is rebuilt with modified xpti headers - all dependent targets are rebuilt.
Previously we had issues without add_custom_command().

Reply:
> project is rebuilt without changes in xpti headers - no dependent targets are built

I think this is not always true with my changes. It always copies headers, and build occurs every second time.

> the project is rebuilt with modified xpti headers - all dependent targets are rebuilt

This is true.

> Previously we had issues without add_custom_command()

Not sure what they were, but we have problem with add_custom_command right now: https://github.com/intel/llvm/pull/4939#issuecomment-972533367

Also locally I see exactly the same issues if I update some file: changes are not picked up by ninja.

Reply:
Discussed offline. Implemented dependency between add_custom_target and add_custom_command with file granularity (instead of directory granularity),